lib/Dialect/AIEVec/Transforms/AIEVectorize.cpp

```diff
@@ -327,6 +327,28 @@ static bool isWellFormedVectorOp(Operation *Op) {
   return true;
 }
 
+// Given an AIEOp, determines if an operation writes to an accumulator
+// based on operation type and operand types
+static bool writesToAccumulator(Operation *op) {
+  // Integer muls and FMAs write to accumulator
+  if (!isAIEOp(op)) {
+    return false;
+  } else if (isa<aievec::MulOp>(op)) {
+    auto mulOp = dyn_cast<aievec::MulOp>(op);
+    auto lhsType = mulOp.lhs().getType().cast<VectorType>().getElementType();
+    auto rhsType = mulOp.rhs().getType().cast<VectorType>().getElementType();
+    return !lhsType.isa<FloatType>() && !rhsType.isa<FloatType>();
```

Comment:
One think that might simplify the logic is to have a different op for floating point operands and for integer operands.  Is there a constraint that the left and right operands are both float or both int?